### PR TITLE
Collect_entries for MatPolyRingZq

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR implements collect_entries for MatPolyRingZq.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 //! Implementations to get information about a [`MatPolynomialRingZq`] matrix.
 
+use flint_sys::{fmpz_poly::fmpz_poly_struct, fmpz_poly_mat::fmpz_poly_mat_entry};
+
 use super::MatPolynomialRingZq;
```
Comment (jnsiemer) [inline src/integer_mod_q/mat_polynomial_ring_zq/get.rs:13]:
```suggestion
use flint_sys::{fmpz_poly::fmpz_poly_struct, fmpz_poly_mat::fmpz_poly_mat_entry};
use super::MatPolynomialRingZq;
```
Remove newline

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs [hunk 2]
```diff
@@ -156,6 +158,41 @@ impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
...
+    /// ```compile_fail
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, ModulusPolynomialRingZq};
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly_mat = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[2  1 2, 3  1 1 1]]").unwrap();
+    /// let poly_ring_mat = MatPolynomialRingZq::from((&poly_mat, &modulus));
+    ///
+    /// let fmpz_entries = poly_ring_mat.collect_entries();
```
Comment (jnsiemer) [inline src/integer_mod_q/mat_polynomial_ring_zq/get.rs:178]: Please add a test in the test module, which executes this doc-test to ensure that it's working properly.
Just enter it under the name `doc_test` s.t. anyone every changing the doc-test can/ should be aware to update this one, too.

File: src/integer_mod_q/mat_polynomial_ring_zq/get.rs [hunk 3]
```diff
@@ -318,3 +355,53 @@ mod test_mod {
...
+        unsafe { fmpz_poly_set(&mut entry2.poly, &entries_1[3]) }
+        unsafe { fmpz_poly_set(&mut entry3.poly, &entries_2[0]) }
+
+        assert_eq!(entries_1.len(), 4);
+        assert_eq!(
+            PolyOverZ::from_str(&format!("1  {}", i64::MAX)).unwrap(),
+            entry1
+        );
+        assert_eq!(
+            PolyOverZ::from_str(&format!("3  {} 1 1", i64::MAX - 58)).unwrap(),
```
Comment (jnsiemer) [inline src/integer_mod_q/mat_polynomial_ring_zq/get.rs:400]: Use `u64::MAX - 58` if it should be a valid prime number. `i64::MAX - 58` isn't prime: 9223372036854775749 = 3×8447×363970326224489 (3 distinct prime factors)

Comment (Marcel583) [reply src/integer_mod_q/mat_polynomial_ring_zq/get.rs:400]: It is not needed to be prime

**Review by jnsiemer — APPROVED** [review]
Please address the remaining issue before merging.
Great work replacing all instances of the previously defined constant!

**Pull request merged**